client/src/User/UserHome/UserInputFoodChoices/UserInputFoodChoices.js

```diff
@@ -146,8 +146,7 @@ class UserInputFoodChoices extends Component {
           this.setState({ businesses: data.jsonBody.businesses });
         })
         .catch(e => console.log(e));
-    } else if(this.state.displayZipMap) {
-        if(this.state.ziplat === 0 && !this.state.notify) {
+    } else if(prevState.displayZipMap !== this.state.displayZipMap && this.state.displayZipMap) {
```

Comment:
I see that you put `else if(prevState.displayZipMap !== this.state.displayZipMap && this.state.displayZipMap)`, isn't that the same as `else if(prevState.displayZipMap !== this.state.displayZipMap)`?

Reply:
If I don't include that last part, then it will generate yelp every time displayZipMap changes, even from true to false. So adding that last part fixed any unnecessary generations. since it should only generate when true.

Reply:
Now you might ask why not just use the last part if it needs to be true. Well then there's an infinite loop of yelp generations as long a displayZipMap is true. So the first part checks if a change occurred to displayZipMap (false, true), and the second makes sure it's actually true.

Reply:
Make sure you check the console and not the page itself. The yelp results will display in the console and not the page whenever the value changes. These results should not have been generated in the first place unless displayZipMap is true.

Reply:
Oh! I didn't check the console that's why. Thanks for pointing that out to me